ambry-network/src/main/java/com.github.ambry.network/Selector.java

```diff
@@ -347,6 +358,24 @@ public void poll(long timeoutMs, List<NetworkSend> sends)
           close(key);
         }
       }
+
+      Iterator<Transmission> sslTransIter = pendingSslHandshakes.iterator();
+      while (sslTransIter.hasNext()) {
+        Transmission sslTransmission = sslTransIter.next();
+        if (sslTransmission != null && sslTransmission.ready()) {
+          connected.add(sslTransmission.getConnectionId());
+          this.metrics.selectorConnectionCreated.inc();
+          Long handshakeStartTime = sslHandshakeTimer.remove(sslTransmission.getConnectionId());
+          if (handshakeStartTime != null) {
+            this.metrics.selectorPerceivedSslHandshakeTime
+                .update(System.currentTimeMillis() - handshakeStartTime.longValue());
+          }
+          sslTransIter.remove();
+        } else if (sslTransmission == null || !keyMap.containsKey(sslTransmission.getConnectionId())) {
```

Comment:
1. Nope. Just to guard against NPE
2. I am update the patch with a fix
3. Synced up offline. will push updates with the fix